Modules/Filtering/MathematicalMorphology/test/itkClosingByReconstructionImageFilterTest.cxx

```diff
@@ -40,75 +40,67 @@ itkClosingByReconstructionImageFilterTest(int argc, char * argv[])
-
   // Declare the type of the Structuring element to be used
   using StructuringElementType = itk::BinaryBallStructuringElement<PixelType, Dimension>;
 
   // Declare the type for the Morphology Filters to be Tested
   using MorphologicalFilterType =
     itk::ClosingByReconstructionImageFilter<InputImageType, OutputImageType, StructuringElementType>;
 
+  using ReaderType = itk::ImageFileReader<InputImageType>;
   ReaderType::Pointer reader = ReaderType::New();
-  WriterType::Pointer writer = WriterType::New();
 
-  // Create the reader and writer
   reader->SetFileName(argv[1]);
-  writer->SetFileName(argv[2]);
+
+  ITK_TRY_EXPECT_NO_EXCEPTION(reader->Update());
+
 
   // Create the filter
   MorphologicalFilterType::Pointer filter = MorphologicalFilterType::New();
 
+  ITK_EXERCISE_BASIC_OBJECT_METHODS(filter, ClosingByReconstructionImageFilter, ImageToImageFilter);
+
+
   StructuringElementType structuringElement;
   structuringElement.SetRadius(std::stoi(argv[3]));
   structuringElement.CreateStructuringElement();
 
   filter->SetKernel(structuringElement);
-  if (std::stoi(argv[4]) == 0)
-  {
-    filter->PreserveIntensitiesOff();
-  }
-  else
-  {
-    filter->PreserveIntensitiesOn();
-  }
+  ITK_TEST_SET_GET_VALUE(structuringElement, filter->GetKernel());
+
+  bool preserveIntensities = static_cast<bool>(std::stoi(argv[4]));
+  ITK_TEST_SET_GET_BOOLEAN(filter, PreserveIntensities, preserveIntensities);
 
-  // Connect the pipelines
   filter->SetInput(reader->GetOutput());
+
+  ITK_TRY_EXPECT_NO_EXCEPTION(filter->Update());
+
+
+  // Write the output
+  using WriterType = itk::ImageFileWriter<OutputImageType>;
+  WriterType::Pointer writer = WriterType::New();
+
+  writer->SetFileName(argv[2]);
   writer->SetInput(filter->GetOutput());
 
-  // Execute print
-  filter->Print(std::cout);
+  ITK_TRY_EXPECT_NO_EXCEPTION(writer->Update());
+
 
-  // Execute the filter
-  try
-  {
-    writer->Update();
-  }
-  catch (const itk::ExceptionObject & excp)
-  {
-    std::cerr << "Exception caught:" << excp << std::endl;
-    return EXIT_FAILURE;
-  }
   // Create a difference image if one is requested
   if (argc == 6)
   {
     itk::SubtractImageFilter<InputImageType, OutputImageType, OutputImageType>::Pointer subtract =
       itk::SubtractImageFilter<InputImageType, OutputImageType, OutputImageType>::New();
     subtract->SetInput(1, reader->GetOutput());
     subtract->SetInput(0, filter->GetOutput());
-    try
-    {
-      writer->SetFileName(argv[5]);
-      writer->SetInput(subtract->GetOutput());
-      writer->Update();
-    }
-    catch (const itk::ExceptionObject & excp)
-    {
-      std::cerr << "Exception caught writing diff image:" << excp << std::endl;
-      return EXIT_FAILURE;
-    }
+
+    writer->SetFileName(argv[5]);
+    writer->SetInput(subtract->GetOutput());
+
+    ITK_TRY_EXPECT_NO_EXCEPTION(writer->Update());
   }
+
+
+  std::cout << "Test finished" << std::endl;
```

Comment:
This line in not really necessary.

If it is really needed and useful, perhaps it could be uniformly added to all tests in "itkTestDriverAfterTest.inc".

Reply:
> it could be uniformly added to all tests in "itkTestDriverAfterTest.inc"

This makes sense. It is somewhat of a big effort to remove all manual instances, though. Perhaps leave it to a new PR?

Reply:
I agree that the line is not necessary, but at the time I was more involved in improving the coverage it was not evident at times to see where or how a test had failed, or if was run at all, reading the standard output in the dashboard. This message allows a human reader to unequivocally know whether the test was run and finished.

If it can be applied to all tests with a single line or few lines in `itkTestDriverAfterTest.inc`, and we are satisfied with the result, then I'd be for adding the necessary code to that file. As Dženan says, it would be best to do it in a separate PR and remove the message present in tests in a single sweep. Maybe an issue can be opened to keep track of it/to avoid forgetting it.